### DISCUSSION TITLE
大量报错,请问哪里出错了

### DISCUSSION THREAD

**Discussion by heheheaa** [XTLS/Xray-core#4081] [Q&A]
2024/11/30 00:25:17 [Warning] [3758843971] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:17 [Warning] [3335206009] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:17 [Warning] [2211762172] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:27 [Warning] [3275114476] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:27 [Warning] [3412914651] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:27 [Warning] [564154515] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:37 [Warning] [1911564706] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:37 [Warning] [1544061098] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"
2024/11/30 00:25:37 [Warning] [1075666661] app/proxyman/inbound: connection ends > proxy/http: failed to read http request > malformed HTTP request "\x00"

配置是最基本的VLESS-XTLS-uTLS-REALITY，同配置之前没问题，升级到24.11.21开始大量出现，回到原版本24.9.30继续存在

**Comment by heheheaa** [comment]
服务器
{``
  "log": {
    "loglevel": "warning",
    "error": "/var/log/xray/error.log",
    "access": "/var/log/xray/access.log"
  },
  "inbounds": [
    {
      "port": 443,
      "protocol": "vless",
      "settings": {
        "clients": [
          {
            "id": "xxxxx", 
            "flow": "xtls-rprx-vision", 
            "email": "xxxxx"
          }
        ],
        "decryption": "none"
      },
      "streamSettings": {
        "network": "tcp",
        "security": "reality",
        "realitySettings": {
          "show": false, 
          "dest": "xxxxx.xxxxx.com:443", 
          "xver": 0, 
          "serverNames": [ 
            "xxxxx.xxxxx.com" 
          ],
          "privateKey": "xxxxx", 
          "shortIds": [ 
            "xxxxx" 
          ]
        }
      },
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ]
      }
    }
  ],
  "routing": {
    "rules": [
      {
        "type": "field",
        "protocol": [
          "bittorrent"
        ],
        "outboundTag": "blocked"
      }
    ]
  },
  "outbounds": [
    {
      "protocol": "freedom",
      "settings": {}
    },
    {
      "tag": "blocked",
      "protocol": "blackhole",
      "settings": {}
    }
  ]
}`

**Comment by heheheaa** [comment]
客户端使用V2rayN

**Comment by heheheaa** [comment]
客户端V2rayN生成的配置
{
  "log": {
    "access": "",
    "error": "",
    "loglevel": "warning"
  },
  "inbounds": [
    {
      "tag": "socks",
      "port": 10808,
      "listen": "127.0.0.1",
      "protocol": "socks",
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ],
        "routeOnly": false
      },
      "settings": {
        "auth": "noauth",
        "udp": true,
        "allowTransparent": false
      }
    },
    {
      "tag": "http",
      "port": 10809,
      "listen": "127.0.0.1",
      "protocol": "http",
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ],
        "routeOnly": false
      },
      "settings": {
        "auth": "noauth",
        "udp": true,
        "allowTransparent": false
      }
    },
    {
      "tag": "socks2",
      "port": 10810,
      "listen": "0.0.0.0",
      "protocol": "socks",
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ],
        "routeOnly": false
      },
      "settings": {
        "auth": "password",
        "udp": true,
        "allowTransparent": false,
        "accounts": [
          {
            "user": "xxxxx",
            "pass": "xxxxx"
          }
        ]
      }
    },
    {
      "tag": "http2",
      "port": 10811,
      "listen": "0.0.0.0",
      "protocol": "http",
      "sniffing": {
        "enabled": true,
        "destOverride": [
          "http",
          "tls"
        ],
        "routeOnly": false
      },
      "settings": {
        "auth": "password",
        "udp": true,
        "allowTransparent": false,
        "accounts": [
          {
            "user": "xxxxx",
            "pass": "xxxxx"
          }
        ]
      }
    }
  ],
  "outbounds": [
    {
      "tag": "proxy",
      "protocol": "vless",
      "settings": {
        "vnext": [
          {
            "address": "xxxxx",
            "port": 443,
            "users": [
              {
                "id": "xxxxx",
                "alterId": 0,
                "email": "xxxxx@xxxxx.xxxxx",
                "security": "auto",
                "encryption": "none",
                "flow": "xtls-rprx-vision"
              }
            ]
          }
        ]
      },
      "streamSettings": {
        "network": "tcp",
        "security": "reality",
        "realitySettings": {
          "serverName": "xxxxx.xxxxx.com",
          "fingerprint": "ios",
          "show": false,
          "publicKey": "xxxxx",
          "shortId": "xxxxx",
          "spiderX": ""
        }
      },
      "mux": {
        "enabled": false,
        "concurrency": -1
      }
    },
    {
      "tag": "direct",
      "protocol": "freedom",
      "settings": {}
    },
    {
      "tag": "block",
      "protocol": "blackhole",
      "settings": {
        "response": {
          "type": "http"
        }
      }
    }
  ],
  "dns": {
    "hosts": {
      "dns.google": "8.8.8.8",
      "proxy.example.com": "127.0.0.1"
    },
    "servers": [
      {
        "address": "223.5.5.5",
        "domains": [
          "geosite:cn"
        ],
        "expectIPs": [
          "geoip:cn"
        ]
      },
      "https://cloudflare-dns.com/dns-query",
      "https://dns.google/dns-query",
      "1.1.1.1",
      "8.8.8.8"
    ]
  },
  "routing": {
    "domainStrategy": "IPIfNonMatch",
    "rules": [
      {
        "type": "field",
        "inboundTag": [
          "api"
        ],
        "outboundTag": "api"
      },
      {
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "domain:example-example.com",
          "domain:example-example2.com"
        ]
      },
      {
        "type": "field",
        "port": "443",
        "network": "udp",
        "outboundTag": "block"
      },
      {
        "type": "field",
        "outboundTag": "block",
        "domain": [
          "geosite:category-ads-all"
        ]
      },
      {
        "type": "field",
        "outboundTag": "direct",
        "domain": [
          "geosite:cn",
          "geosite:geolocation-cn"
        ]
      },
      {
        "type": "field",
        "outboundTag": "direct",
        "ip": [
          "223.5.5.5/32",
          "119.29.29.29/32",
          "180.76.76.76/32",
          "114.114.114.114/32",
          "geoip:private",
          "geoip:cn"
        ]
      },
      {
        "type": "field",
        "port": "0-65535",
        "outboundTag": "proxy"
      }
    ]
  }
}

**Comment by RayneScofield** [comment]
What's happening?

**Comment by southBeforeNorth** [comment]
请问你找到原因了吗，我也有类似的报错，手机的小火箭可以正常使用，但是同样的配置，v2rayN（v7.4.1）报这个错

**Comment by janesle** [comment]
没人回复啊，有一个Q群讨论：939989248

**Comment by Godsix** [comment]
这个应该是sock5和http二合一的问题，监听同一个端口，根据协议内容来判断是sock5还是http。

**Comment by buxh2017** [comment]
我也遇到了这个问题，点击v2rayN的检查更新，将它的依赖组件更新后就好了
